bin/create-release

```diff
@@ -0,0 +1,197 @@
+
+if not os.path.exists("gh-release-token.txt"):
+    print("gh-release-token.txt not found in root directory", file=sys.stderr)
+    print(
+        "Create one with repo and read:org scopes at https://docs.github.com/en/enterprise-server@3.4/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token",
+        file=sys.stderr)
+    exit(1)
+
+if len(sys.argv) != 3:
+    print(
+        "Usage: bin/create-release COMMIT_SHA RELEASE_NUMBER", file=sys.stderr)
+    exit(1)
+
+
+def shell_cmd(command_string):
+    try:
+        process = subprocess.run(
+            shlex.split(command_string), capture_output=True)
+    except FileNotFoundError:
+        return ''
+    return process.stdout.decode('ascii') if process else ''
+
+
+################################################################################
+# Resolve and validate GitHub token, check auth
+################################################################################
+
+with open("gh-release-token.txt", "r") as file:
+    GH_TOKEN = file.read().rstrip()
+
+if not GH_TOKEN.startswith("ghp_"):
+    print("Invalid token found in gh-release-token.txt", file=sys.stderr)
+    exit(1)
+
+check_gh_auth_response = requests.get(
+    "https://api.github.com/repos/seattle-uat/civiform",
+    headers={"Authorization": "token " + GH_TOKEN})
+if check_gh_auth_response.status_code != 200:
+    print(
+        "GitHub API auth failed, satus code: " +
+        str(check_gh_auth_response.status_code),
+        file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Check docker hub auth
+################################################################################
+
+if not "Login Succeeded" in shell_cmd(
+        "docker login") or not "civiform" in shell_cmd("docker info"):
+    print("Unable to login to docker hub")
+    exit(1)
+
+################################################################################
+# Set release variables
+################################################################################
+
+COMMIT_SHA = sys.argv[1]
+RELEASE_NUMBER = sys.argv[2]
+RELEASER_NAME = shell_cmd("git config --get user.name")
+RELEASER_EMAIL = shell_cmd("git config --get user.email")
+
+################################################################################
+# Validate release version number
+################################################################################
+
+if not re.match("^v\d+\.\d+\.\d+$", RELEASE_NUMBER):
+    print("Invalid version number: " + RELEASE_NUMBER, file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Validate commit SHA
+################################################################################
+
+branches_containing_commit_sha = shell_cmd(
+    "git --no-pager branch --contains " + COMMIT_SHA)
+branches_containing_commit_sha = [
+    re.sub('\*', '', x).strip()
```

Comment:
Blocking: This regular expression is non-obvious. I'd add a quick comment indicating sample format to make it easier to determine what this is actually substituting.

Reply:
done

Reply:
Thanks!